### ISSUE TITLE
Faulty pyproject.toml generated by poetry-lock-package

### ISSUE THREAD

**Issue by jeroeness** [bneijt/poetry-lock-package#36]
# Problem description

this program produces a folder "..._lock" when executed: `poetry run poetry-lock-package --build`. This folder contains a faulty `pyproject.toml` since poetry is unable to parse it correctly:
```
$ Poetry lock
Updating dependencies
Resolving dependencies... (10.4s)
$ poetry run poetry-lock-package
INFO: Writing common-lock
$ cd common-lock
$ poetry lock
Could not parse version constraint: markers
```

# Symptoms
When I run `poetry run poetry-lock-package --build`, I get an error. There seems to be something strange going on with generating the lock package. outputs shown here: https://pastebin.com/9hbrygwE

It boils down to `poetry build --format wheel` being called by https://github.com/bneijt/poetry-lock-package/blob/main/poetry_lock_package/app.py#L244 in the "....-lock" directory crashing because it encounters a package version that is incorrect. Normally, a package version is something like `^1.2.3`, but somehow a version `markers` is coming up now. I have no idea how that can happen. It is possible that Poetry has been updated further while `poetry-lock-package` has not been updated.

# Reproduce
To reproduce, take my [pyproject.toml.txt](https://github.com/bneijt/poetry-lock-package/files/15482947/pyproject.toml.txt), poetry lock and install packages, run `poetry-lock-package --build`
tested on python version 3.10 & 3.9
poetry version 1.8.3

**Comment by bneijt** [comment]
Bug indeed, closed in version 0.5.1. The issue arose because the markers we try to collect for urllib came from multiple version constraints in dependency listings.

The best fix would be to match the version constraint to find the markers that should be applied to the locked version, but we can't guarantee that that makes sense. This will only become an issue if we run into a package having different platform markers, for example, for different version ranges (so if package A depends on package B v1 if platform is windows and package B v2 if platform is Linux, the lock chooses the version and therefore enforces the marker, but we currently do not support that as it's very unlikely to happen).

Long story short, 1.5.1 should fix this issue and thank you very much for taking the time to report this issue!
